src/nbody_to_ode.jl

```diff
@@ -277,4 +356,34 @@ function gather_accelerations_for_potentials(simulation::NBodySimulation{<:Water
     push!(h_acelerations, get_accelerating_function(simulation.system.scpfw_parameters, simulation))
 
     (o_acelerations, h_acelerations)
+end
+
+function DiffEqBase.SDEProblem(simulation::NBodySimulation{<:PotentialNBodySystem})
+    (u0, v0, n) = gather_bodies_initial_coordinates(simulation)
+    
+    acceleration_functions = gather_accelerations_for_potentials(simulation)   
+
+    therm = simulation.thermostat
+    m = simulation.system.bodies[1].m
+
+    function f!(du, u, p, t)
+        du[:, 1:n] = @view u[:, n + 1:2n];
+
+        @inbounds for i = 1:n
+            a = MVector(0.0, 0.0, 0.0)
+            for acceleration! in acceleration_functions
+                acceleration!(a, u[:, 1:n], u[:, n + 1:end], t, i);                
+            end
+            du[:, n + i] .= a
+        end
+        
+        @. du[:, n + 1:end] -= therm.γ*u[:, n + 1:end]/m
+    end
+
+    function g!(du, u, p, t)
+        du[:, n + 1:end] += sqrt(2*therm.γ*simulation.kb*therm.T)/m
```

Comment:
I think you need to `@.` this to have it not allocate?

Reply:
it wasn't obvious.... but yeah, right.